benchmarks/graphics_pipeline/main.cpp

```diff
@@ -743,6 +847,74 @@ void ProjApp::Setup()
     }
 }
 
+void ProjApp::RepeatGeometryNonPositionVertexData(Geometry& dstGeom, const Geometry& srcGeom, size_t repeatCount)
+{
+    size_t nVertexBufferCount = srcGeom.GetVertexBufferCount();
+    PPX_ASSERT_MSG(nVertexBufferCount == dstGeom.GetVertexBufferCount(), "Mismatched source and destination vertex data format");
+
+    bool copyFirstBuffer;
+    switch (nVertexBufferCount) {
+        case 0: {
+            PPX_ASSERT_MSG(0, "Geometry cannot have 0 vertex buffers");
+        } break;
+        case 1: {
+            // For interleaved (1 vb), repeat position data as well
+            copyFirstBuffer = true;
+        } break;
+        default: {
+            // For position planar (2 vb) & planar (1+ vb), repeat only non-position vertex data
+            copyFirstBuffer = false;
+        }
+    }
+
+    for (size_t vertexBufferIndex = 0; vertexBufferIndex < nVertexBufferCount; vertexBufferIndex++) {
+        if (vertexBufferIndex == 0 && !copyFirstBuffer) {
+            continue;
+        }
+
+        const char*      srcBufferDataPtr = srcGeom.GetVertexBuffer(vertexBufferIndex)->GetData();
+        size_t           srcBufferSize    = srcGeom.GetVertexBuffer(vertexBufferIndex)->GetSize();
+        size_t           dstBufferSize    = srcBufferSize * repeatCount;
+        Geometry::Buffer dstBuffer        = *(dstGeom.GetVertexBuffer(vertexBufferIndex));
```

Comment:
probably Geometry::Buffer& to avoid an extra copy of dstBuffer?

Reply:
Obsolete due to setter changes